### PR TITLE
fix: datepicker blocks input field

### PR DESCRIPTION
**Pull request by MarcoKorinth**
# Changes
- fixed bug: datepicker opens over input field
- datepicker only opens after clicking the new calendar icon inside the input field
- datepicker starts in year selection mode by default

### CODE REVIEW

File: frontend/src/js/common/helpers/mergeRefs.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+type Mutable<T> = {
+  -readonly [k in keyof T]: T[k];
+};
```
Comment (Kadrian) [inline frontend/src/js/common/helpers/mergeRefs.ts:3]: Hmmm in other implementations I can see them using a `MutableRefObject` type from React. You probably copied the code below from somewhere, right? I guess I wonder whether we shouldn't just use some external implementation like https://github.com/gregberge/react-merge-refs but up to you

**Review by Kadrian — CHANGES_REQUESTED** [review]
I'd be happy if you tried reusing `IconButton`, otherwise looks pretty good!

**Review by Kadrian — APPROVED** [review]
Nice!
Slightly funny that the year picker half way overlays the day picker if there aren't that many years. But I hope that will just be fixed in the backend with more available years, cc @awildturtok 

Here's a small suggestion https://github.com/ingef/conquery/pull/3166

**Comment by awildturtok** [comment]
@MarcoKorinth kannst du bitte die dependencies auf eva seite gleichstellen?

**Pull request merged**